src/wasm2asm.h

```diff
@@ -578,6 +583,49 @@ void Wasm2AsmBuilder::addExports(Ref ast, Module* wasm) {
   ast->push_back(ValueBuilder::makeStatement(ValueBuilder::makeReturn(exports)));
 }
 
+void Wasm2AsmBuilder::addGlobal(Ref ast, Global* global) {
+  assert(global->init);
+
+  if (global->init->is<Const>()) {
+    auto const_ = static_cast<Const*>(global->init);
+    switch (const_->type) {
+      case Type::i32: {
+        Ref theVar = ValueBuilder::makeVar();
```

Comment:
in the 3 cases, the only difference is the actual value, all the rest (makeVar, etc.) is the same. How about in the switch cases just doing the different stuff, and sharing the rest before & after the switch?